DOCS: Clarification on the use of `label_names` as an argument to TrainingArguments

huzaifa-jawad367 opened the pull request:
# What does this PR do?

This PR clarifies the documentation for the label_names argument in TrainingArguments. Currently, it may be unclear when label_names should be used. For classification tasks (binary, multi-class), labels are expected as a single tensor (labels), so label_names is not required and can cause errors if set. label_names is only relevant for tasks where models accept multiple label arguments (e.g., QA).


Fixes #40217

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. (Fixes #40217)
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests? (Not applicable, doc-only PR)


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@zach-huggingface @stevhliu

huzaifa-jawad367 committed:
Update trainer.md

huzaifa-jawad367 committed:
Merge branch 'huggingface:main' into fix-metric-for-best-model-f1

stevhliu reviewed (commented):
Maybe it'd be better to clarify the actual docstring [here](https://huggingface.co/docs/transformers/en/main_classes/trainer#transformers.TrainingArguments.label_names). It can say something like:

```py
label_names (list[str], optional):
   The list of keys in your dictionary of inputs that correspond to the labels. Label names are automatically detected from a model's `forward` method by looking for `"label"` or `"start_positions"` and `"end_positions"` for question answering models.

   You should only specify `label_names` if you're using custom label names or multiple labels.
```

huzaifa-jawad367 committed:
Merge branch 'main' into fix-metric-for-best-model-f1

huzaifa-jawad367 committed:
Update trainer.md

Removed the detail about label_names argument usage from the tip/ warning section

huzaifa-jawad367 committed:
Update training_args.py

Added the label_names usage clarification in the docstring

stevhliu committed:
Update trainer.md

stevhliu reviewed (approved):
Thanks!
